### PR TITLE
More Slime Reactions

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Using blood on an Oil Slime will create Corn Oil.
add: Using blood on Pyrite Slimes will create a random crayon.
add: Using water on Orange Slimes will create a small smoke cloud.
add: Using water on Blue Slimes will spawn foam.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 1]
```diff
@@ -233,6 +233,17 @@
...
+/datum/chemical_reaction/slimefoam
+	name = "Slime Foam"
+	id = "m_foam"
+	results = list("fluorosurfactant" = 20, "water" = 20)
+	required_reagents = list("water" = 5)
+	required_container = /obj/item/slime_extract/blue
+	required_other = 1
+
+/datum/chemical_reaction/slimefoam/on_reaction(datum/reagents/holder)
+		feedback_add_details("slime_cores_used","[type]")
```
Comment (phil235) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]: two tabs?

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 2]
```diff
@@ -300,6 +311,18 @@
...
+/datum/chemical_reaction/slimesmoke
+	name = "Slime Smoke"
+	id = "m_smoke"
+	results = list("phosphorus" = 10, "potassium" = 10, "sugar" = 10)
+	required_reagents = list("water" = 5)
+	required_container = /obj/item/slime_extract/orange
+	required_other = 1
+
+/datum/chemical_reaction/slimesmoke/on_reaction(datum/reagents/holder)
+		feedback_add_details("slime_cores_used","[type]")
```
Comment (phil235) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]: again

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 3]
```diff
@@ -627,6 +661,22 @@
...
+	name = "Slime Crayon"
+	id = "s_crayon"
+	required_reagents = list("blood" = 1)
+	required_container = /obj/item/slime_extract/pyrite
+	required_other = 1
+
+/datum/chemical_reaction/slimecrayon/on_reaction(datum/reagents/holder)
+	feedback_add_details("slime_cores_used","[type]")
+	var/list/crayons = subtypesof(/obj/item/toy/crayon)
+	var/chosen = pick(crayons)
```
Comment (nullbear) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]: this has a chance to create a spraycan iirc. Should probably uses a weighted_pick() list.

**Comment by XDTM** [comment]
Turns out gangs were using spraycan-shaped crayons all along

**Comment by pudl** [comment]
Dont include a changelog for this

**Comment by XDTM** [comment]
It's a feature, if i don't put a changelog only those that browse github will use it

**Pull request merged**